content/en/docs/guides/server-graceful-stop.md

```diff
@@ -0,0 +1,86 @@
+---
+title: GracefulStop
+description: >-
+  Explains how to gracefully shut down a gRPC server.
+---
+
+### Overview
+
+gRPC servers often need to shut down gracefully, ensuring that in-flight RPCs
+are completed within a reasonable timeframe and new RPCs are no longer
+accepted. `GracefulStop()` facilitates this process, allowing the server to
+transition smoothly without abruptly terminating active connections.
+
+When `GracefulStop()` is called, the server immediately stops accepting new
+RPCs. In-flight RPCs are allowed to continue until they complete or a specified
+deadline is reached. Once all active RPCs finish or the deadline expires, the
+server shuts down completely.
+
+For detailed semantics see [this][grpc doc].
+
+### How to use Wait-for-Ready
+
+The exact implementation of `GracefulStop()` varies depending on the
+programming language you are using. However, the general pattern
+involves:
+
+- Initiating the graceful shutdown process by calling `GracefulStop()` on your
+  gRPC server object. This function blocks until all currently running RPCs
+  completed. This ensures that in-flight requests are allowed to finish
+  processing.
+- Specify a timeout period to limit the time allowed for in-progress RPCs to
+  finish. It's crucial to call `Stop()` on server object with a timeout before
+  calling `GracefulStop()`. This acts as a safety net, ensuring that the server
+  eventually shuts down even if some in-flight RPCs don't complete within a
+  reasonable timeframe.  This prevents indefinite blocking.
+
+The following shows the sequence of events that occur, when a server graceful
+shutdown is invoked while active RPCs are being process and client is sending
+new RPCs.
```

Comment:
Can you also mention here the two alternative scenarios the diagram covers?

Reply:
Added